### PR TITLE
Document extraMounts and extraVol

### PR DESCRIPTION
**Pull request by fultonj**
Document extraMounts and extraVol

Co-authored-by: Francesco Pantano - fpantano@redhat.com

### CODE REVIEW

**Review by fmount — COMMENTED** [review]
I did a first pass on it (Thanks John for the help) and I left a few comments on things we can improve on a next iteration

File: images/extra_mounts_propagation.png
Comment (fmount) [inline images/extra_mounts_propagation.png:1]: Remove

File: images/extra_mounts_struct.png
Comment (fmount) [inline images/extra_mounts_struct.png:1]: Remove

File: extra_mounts.md [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+The following is an explanation of the `volMounts` struct in
+[modules/storage/storage.go](https://github.com/openstack-k8s-operators/lib-common/blob/main/modules/storage/storage.go).
+
+![ExtraMounts Struct](images/extra_mounts_struct.png)
+
+Examples:
+
+- The `ExtraVolType` label can be used in the logic of an operator: 
+[dataplane-operator usage](https://github.com/openstack-k8s-operators/dataplane-operator/blob/5ebd8ad49a6b674c930b24f28f3da4656ac088ef/pkg/deployment/deployment.go#L224-L229).
+```
```
[the remark this answers is not in the document]
Comment (fultonj) [reply extra_mounts.md:87]: I found `go` not `golang` worked in my local markdown env so I used that. It will be in the next ps.

**Comment by fultonj** [comment]
@fmount Thanks for the ascii art I've replaced the png fileswith it.

**Review by fmount — APPROVED** [review]

**Pull request merged**